version: '3.7'

services:
  # Backend
  api:
    container_name: test-api
    restart: unless-stopped
    build:
      context: .
      dockerfile: Dockerfile.dev
    ports:
      - '${APP_PORT:-3000}:3000'
    networks:
      - api_network
    volumes:
      - ./:/var/www/test-task-api
    environment:
      NODE_ENV: development
    depends_on:
      - postgres-db

  # Database
  postgres-db:
    container_name: db-postgres
    restart: unless-stopped
    image: postgres:14.8-alpine3.18
    environment:
      TZ: 'Europe/Moscow'
      POSTGRES_DB: '${DB_NAME:-test-db}'
      POSTGRES_USER: '${DB_USER:-root}'
      POSTGRES_PASSWORD: '${DB_PASSWORD:-root}'
    healthcheck:
      test: ['CMD-SHELL', 'pg_isready -U ${DB_USER:-root} -d ${DB_NAME:-test-db}']
      interval: 10s
      timeout: 5s
      retries: 5
      start_period: 10s
    ports:
      - '5432:5432'
    volumes:
      - pg_data:/var/lib/postgresql/data
    networks:
      - api_network

  # Web GUI pg
  pgadmin:
    container_name: pgadmin_container
    image: dpage/pgadmin4:7.2
    environment:
      PGADMIN_DEFAULT_EMAIL: 'root@root.com'
      PGADMIN_DEFAULT_PASSWORD: 'root'
      PGADMIN_CONFIG_SERVER_MODE: 'False'
    volumes:
      - docker-pgadmin:/var/lib/pgadmin
    ports:
      - '5051:80'
    restart: unless-stopped
    depends_on:
      - postgres-db
    networks:
      - api_network

volumes:
  pg_data:
  docker-pgadmin:

networks:
  api_network:
    external: true
